src/hdtXMLImprover.cpp

```
@@ -0,0 +1,132 @@
+namespace hdt
+{
+	using ChildSet = std::unordered_set<std::string>;
+	using ChildMap = std::unordered_map<std::string, ChildSet>;
+
+	// ---- DOM walker ----
+
+	// Recursively removes child elements that are either:
+	//   (a) not in knownElements — completely unknown tag, or
+	//   (b) not in the allowed children set for their parent.
+	// Returns true if any element was removed at this level or below.
+	static bool cleanNode(
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)